### PR TITLE
Solution Operator role manifest

### PR DESCRIPTION
**Pull request by alexandre-allard**
**Component**: salt, solutions, examples

**Context**: 

**Summary**:

**Acceptance criteria**: 

Waiting for #2565 to be merged.

---

Closes: #2389

### CODE REVIEW

**Review by gdemonet — COMMENTED** [review]
LGTM, maybe just add details in the Solution Operator guidelines for how this will work (especially the error case if something else than `Role`/`ClusterRole` objects are defined in the file). Would make sense to provide an easy tool for validation of an ISO (outside of a running MetalK8s cluster).

File: docs/developer/solutions/operator.rst [hunk 1]
```diff
@@ -105,3 +105,7 @@ directory. This file is a manifest which declares all necessary ``Role`` and
 MetalK8s will take care of deploying these objects, create a ``ServiceAccount``
 named ``<solution_name>-operator`` and all needed ``RoleBinding`` to bind these
 roles to this account.
+
+.. warning:: Only ``Role`` and ``ClusterRole`` kinds are allowed in this file,
+             the deployment of the Solution fail if any other resource is
+             found.
```
Comment (gdemonet) [inline docs/developer/solutions/operator.rst]: Would be best written like:

```suggestion
.. warning::

   Only ``Role`` and ``ClusterRole`` kinds are allowed in this file,
   the deployment of the Solution fail if any other resource is found.
```

File: salt/metalk8s/orchestrate/solutions/files/operator/role_binding.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 kind: RoleBinding
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
-  name: {{ solution }}-operator
+  name: {{ solution }}-operator-{{ role_kind }}-{{ role_name }}
```
Comment (gdemonet) [inline salt/metalk8s/orchestrate/solutions/files/operator/role_binding.yaml]:
```suggestion
  name: {{ solution }}-operator-{{ role_kind | lower }}-{{ role_name }}
```

**Review by gdemonet — DISMISSED** [review]

File: docs/developer/solutions/operator.rst [hunk 3]
```diff
@@ -105,3 +105,8 @@ directory. This file is a manifest which declares all necessary ``Role`` and
 MetalK8s will take care of deploying these objects, create a ``ServiceAccount``
 named ``<solution_name>-operator`` and all needed ``RoleBinding`` to bind these
 roles to this account.
+
+.. warning::
+
+   Only ``Role`` and ``ClusterRole`` kinds are allowed in this file,
+   the deployment of the Solution fail if any other resource is found.
```
Comment (Ebaneck) [inline docs/developer/solutions/operator.rst]: Minor
```suggestion
   Only ``Role`` and ``ClusterRole`` kinds are allowed in this file,
   the deployment of the Solution fails if any other resource is found.
```

**Review by Ebaneck — APPROVED** [review]
Couldn't successfully run Post-merge on this!
Let's merge it.

**Pull request merged**